src/model/Board.java

```diff
@@ -89,6 +119,18 @@ public Square getSquare(int position) {
         return squares;
     }
 
+    public Set<Square> getSquaresForPlayer(PieceColor color) {
+        Set<Square> playersSquares = new HashSet<>();
```

Comment:
A HashSet? Interesting! I haven't used one of these before. Perhaps a quick explanation of the choice? Just curious. :)